src/khoj/routers/api.py

```diff
@@ -686,16 +686,20 @@ def edit_job(
 
     # Check, get automation to edit
     try:
-        automation: Job = AutomationAdapters.get_automation(user, automation_id)
+        automation: Job = await AutomationAdapters.aget_automation(user, automation_id)
```

Comment:
Why not write a corresponding synchronous version of the `aget_automation` function and wrap the `schedule_query` function in a async to sync method so that this API can remain synchronous?